source/utils/Random.cpp

```diff
@@ -36,58 +30,35 @@
 unsigned long myRandomSeed;
 const unsigned long MAX = 32768;
 
-unsigned long randgen()
+static unsigned long randgen()
 {
     myRandomSeed = myRandomSeed * 1103515245 + 12345;
-    unsigned long returnVal = (unsigned int) (myRandomSeed / 65536) % MAX;
+    //TODO: What is the purpose of the cast?
```

Comment:
Then, IIRC, when A / B, B is promoted to the type of A. This means 65536 is promoted as an unsigned long and maybe the cast does put this back to an unsigned int afterwards to play fine with ```% MAX.``` or have better computation performance maybe?
Otherwise, I have no idea why.